src/frontend-samples/scientific-visualization-sample/AnimationWidget.tsx

```diff
@@ -0,0 +1,98 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import React, { useEffect } from "react";
+import { Select } from "@bentley/ui-core";
+import { AbstractWidgetProps, StagePanelLocation, StagePanelSection, UiItemsProvider, WidgetState } from "@bentley/ui-abstract";
+import { StandardViewId } from "@bentley/imodeljs-frontend";
+import "./AnimationWidget.scss";
+import AnimationApi from "./AnimationApi";
+import { AnalysisDecorator } from "./AnalysisDecorator";
+import { useActiveViewport } from "@bentley/ui-framework";
+const meshTypes = ["Cantilever", "Flat with waves"];
+
+export const AnimationWidget: React.FunctionComponent = () => {
+  const [meshType, setMeshType] = React.useState<{ meshPicker: string, stylePicker: string[], defaultStylePicker: string }>({ meshPicker: "Cantilever", stylePicker: [], defaultStylePicker: "" });
+  const viewport = useActiveViewport();
+
+  const renderAnimation = async (meshingType: string) => {
+    if (viewport) {
+      const meshes = await Promise.all([
+        AnimationApi.createMesh("Cantilever", 100),
+        AnimationApi.createMesh("Flat with waves"),
+      ]);
+      if (AnalysisDecorator.decorator) {
+        viewport.displayStyle.settings.analysisStyle = undefined;
+        AnalysisDecorator.decorator.dispose();
+      }
+      const mesh = meshingType === "Cantilever" ? meshes[0] : meshes[1];
+      AnalysisDecorator.decorator = new AnalysisDecorator(viewport, mesh);
+      const meshStyle: string[] = [];
+      for (const name of AnalysisDecorator.decorator.mesh.styles.keys()) {
+        meshStyle.push(name);
+      };
+      setMeshType({ meshPicker: meshingType, stylePicker: [...meshStyle], defaultStylePicker: meshStyle[0] });
+    }
+  };
+
+  useEffect(() => {
```

Comment:
We typically initialize viewstate in the app.tsx file using the viewStateOptions prop.  Please move this to that file.

Reply:
Since i have used BlankConnectionViewState and it does not have all props of ViewFlags, so keeping it in the same file for now